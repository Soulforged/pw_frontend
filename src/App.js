//@flow
import React from "react";
import { Provider, connect } from "react-redux";
import "src/App.css";
import { ErrorBoundary } from "src/error";
import { Login } from "src/session";
import { Home } from "src/home";
import { Loading } from "src/components";
import { ConnectedRouter } from "react-router-redux";
import { Route, Switch } from "react-router-dom";
import SecureRoute from "src/SecureRoute";
import createStore from "src/reduxconf";
import createHistory from "history/createBrowserHistory";

import type { Store } from "src/types";

type Props = {
  store?: Store<*, *>,
  history?: Object
};

const ThemeLinkC = ({ dark }: { dark: boolean }) => (
  dark ? <link rel="stylesheet" type="text/css" href="/AppDark.css" /> : false
);

const ThemeLink = connect(({ ui: { dark } }) => ({ dark }))(ThemeLinkC);

const App = ({ store, history }: Props) => (
  <Provider store={store}>
    <div>
      <ThemeLink />
      <ErrorBoundary>
        <ConnectedRouter history={history}>
          <div>
            <Switch>
              <Route path="/login" component={Login} />
              <SecureRoute path="/" component={Home} />
            </Switch>
            <Loading />
          </div>
        </ConnectedRouter>
      </ErrorBoundary>
    </div>
  </Provider>
);

const defaultHistory = createHistory();
App.defaultProps = {
  history: defaultHistory,
  store: createStore({ history: defaultHistory, logActions: process.env.DEBUG_APP })
};

export default App;
