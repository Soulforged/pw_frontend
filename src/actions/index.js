//@flow
import { createActions } from "redux-actions";
import { CALL_API } from "src/constants";
import type { UserInfo, AppError } from "src/types";

const ENTITIES = "@@entities";
const SAVE = "SAVE";
const REQ = "REQUEST";
const SUCCESS = "SUCCESS";
const FAIL = "FAILURE";
const LOGIN_SUCCESS = "LOGIN_SUCCESS";
const LOGIN_REQUEST = "LOGIN_REQUEST";
const LOGIN_FAILURE = "LOGIN_FAILURE";

const saveTypes = (actionName) => {
  const normActionName = actionName.toUpperCase();
  return [
    `${ENTITIES}/${SAVE}_${normActionName}_${REQ}`,
    `${ENTITIES}/${SAVE}_${normActionName}_${SUCCESS}`,
    `${ENTITIES}/${SAVE}_${normActionName}_${FAIL}`
  ];
};

const fetchTypes = actionName => [
  `${actionName}_${REQ}`,
  `${actionName}_${SUCCESS}`,
  `${actionName}_${FAIL}`
];

const fetchEntitiesTypes = actionName => fetchTypes(`${ENTITIES}/${actionName.toUpperCase()}`);

const login = ({ username, password }: UserInfo) => ({
  [CALL_API]: {
    types: [LOGIN_REQUEST, LOGIN_SUCCESS, LOGIN_FAILURE],
    endpoint: "/authenticate/generate-token",
    body: { user: username, password }
  }
});

const syncActions = createActions({
  SET_ERROR: (error: AppError) => error,
  LOGOUT: () => ({}),
  TOGGLE_THEME: () => ({}),
  TOGGLE_MENU: () => ({}),
  TOGGLE_USER_MENU: () => ({}),
});

export default {
  ENTITIES,
  SAVE,
  REQ,
  SUCCESS,
  FAIL,
  saveTypes,
  fetchTypes,
  fetchEntitiesTypes,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGIN_REQUEST,
  login,
  ...syncActions
};
