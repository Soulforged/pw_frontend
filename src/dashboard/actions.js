//@flow
import { CALL_API } from "src/constants";
import moment from "moment";
import mocks from "src/mocks";

export const DASHBOARD_SUMMARY_SUCCESS = "DASHBOARD_SUMMARY_SUCCESS";
export const DASHBOARD_TRENDS_BY_DATE_SUCCESS = "DASHBOARD_TRENDS_BY_DATE_SUCCESS";
export const DASHBOARD_TRENDS_BY_INSTITUTION_SUCCESS = "DASHBOARD_TRENDS_BY_INSTITUTION_SUCCESS";
export const DASHBOARD_ERRORS_SUCCESS = "DASHBOARD_ERRORS_SUCCESS";

const now = moment();
const dateTo = now.format("YYYY-MM-DD");
const dateFrom = now.subtract(7, "days").format("YYYY-MM-DD");
const defaultParams = { dateFrom, dateTo };

const params = prms => ({ ...defaultParams, ...prms });

export const fetchSummary = prms => ({
  [CALL_API]: {
    types: ["DASHBOARD_SUMMARY_REQUEST", DASHBOARD_SUMMARY_SUCCESS, "DASHBOARD_SUMMARY_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params(prms).dateFrom}&date_to=${params(prms).dateTo}&distinct=sender_account_number`,
    //FIXME remove this mock
    mock: mocks.summary,
    key: "summary"
  }
});

export const fetchTrendsByDate = prms => ({
  [CALL_API]: {
    types: ["DASHBOARD_TRENDS_BY_DATE_REQUEST", DASHBOARD_TRENDS_BY_DATE_SUCCESS, "DASHBOARD_TRENDS_BY_DATE_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params(prms).dateFrom}&date_to=${params(prms).dateTo}&group_by=date`,
    mock: mocks.trendsByDate,
    key: "trendsByDate"
  }
});

export const fetchTrendsByInstitution = prms => ({
  [CALL_API]: {
    types: ["DASHBOARD_TRENDS_BY_INSTITUTION_REQUEST", DASHBOARD_TRENDS_BY_INSTITUTION_SUCCESS, "DASHBOARD_TRENDS_BY_INST_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params(prms).dateFrom}&date_to=${params(prms).dateTo}&group_by=institution_name`,
    mock: mocks.trendsByInstitution,
    key: "trendsByInstitution"
  }
});

export const fetchErrors = prms => ({
  [CALL_API]: {
    types: ["DASHBOARD_ERRORS_REQUEST", DASHBOARD_ERRORS_SUCCESS, "DASHBOARD_ERRORS_FAILURE"],
    endpoint: `/bi/totals/transactions?date_from=${params(prms).dateFrom}&date_to=${params(prms).dateTo}&group_by=status&status=ERROR`,
    key: "errors"
  }
});
