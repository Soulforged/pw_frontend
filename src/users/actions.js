//@flow
import { CALL_API } from "src/constants";
import mocks from "src/mocks";
import { user, list, error } from "./schemas";

export const fetchUsers = () => ({
  [CALL_API]: {
    types: ["@@entities/USERS_REQUEST", "@@entities/USERS_SUCCESS", "@@entities/USERS_FAILURE"],
    endpoint: "/user/backofficeuser",
    schema: list(user),
    errorSchema: error,
    key: "users",
  }
});

export const fetchUserByCriteria = ({ criteria, filter }) => {
  if (!filter) {
    return fetchUsers();
  }
  return {
    [CALL_API]: {
      types: ["@@entities/USERS_FILTER_REQUEST", "@@entities/USERS_FILTER_SUCCESS", "@@entities/USERS_FILTER_FAILURE"],
      endpoint: `/user/backofficeuser/${criteria}/${filter}`,
      schema: user,
      errorSchema: error,
      key: "users",
      mock: mocks.users
    }
  };
};
