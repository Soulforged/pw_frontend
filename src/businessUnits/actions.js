//@flow
import { push } from "react-router-redux";
import { CALL_API } from "src/constants";
import actions from "src/actions";
import { businessUnit, list, error } from "./schemas";

const { fetchEntitiesTypes, saveTypes } = actions;

const BUSINESSUNIT = "BUSINESSUNIT";
const BUSINESSUNITS = "BUSINESSUNITS";

const restPath = body => (body.id ? `/${body.id}` : "");

export const saveBusinessUnit = (body: Object) => ({
  [CALL_API]: {
    types: saveTypes(BUSINESSUNIT),
    endpoint: `/user/businessunit${restPath(body)}`,
    body,
    update: body.id != null,
    key: "businessUnits",
    errorSchema: error,
    after: dispatch => dispatch(push("/business_units"))
  }
});

const fetchAllBusinessUnits = () => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(BUSINESSUNITS),
    endpoint: `/user/businessunit?companyId=1&companyType=Merchant`,
    schema: list(businessUnit),
    errorSchema: error,
    key: "businessUnits",
    invalidatesCache: true
  }
});

export const fetchBusinessUnits = ({ name }: { name: string } = {}) => {
  if (!name) {
    return fetchAllBusinessUnits();
  }
  return ({
    [CALL_API]: {
      types: fetchEntitiesTypes(BUSINESSUNITS),
      endpoint: `/user/businessunit?companyId=1&companyType=Merchant&name=${name}`,
      schema: list(businessUnit),
      errorSchema: error,
      key: "businessUnits",
      invalidatesCache: true
    }
  });
};

export const getBusinessUnit = (id: number) => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(BUSINESSUNIT),
    endpoint: `/user/businessunit/${id}`,
    schema: businessUnit,
    errorSchema: error,
    key: "businessUnits"
  }
});
