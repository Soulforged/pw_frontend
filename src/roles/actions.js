//@flow
import { push } from "react-router-redux";
import { CALL_API } from "src/constants";
import actions from "src/actions";
import { role, list, error } from "./schemas";

const { fetchEntitiesTypes, saveTypes } = actions;

const ROLE = "ROLE";
const ROLES = "ROLE";

const restPath = body => (body.id ? `/${body.id}` : "");

export const saveRole = (body: Object) => ({
  [CALL_API]: {
    types: saveTypes(ROLE),
    endpoint: `/user/role${restPath(body)}`,
    body,
    update: body.id != null,
    key: "roles",
    errorSchema: error,
    after: dispatch => dispatch(push("/roles"))
  }
});

export const fetchRole = id => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(ROLE),
    endpoint: `/user/role/${id}`,
    schema: role,
    errorSchema: error,
    key: "roles"
  }
});

export const fetchRoles = () => ({
  [CALL_API]: {
    types: fetchEntitiesTypes(ROLES),
    endpoint: "/user/role",
    schema: list(role),
    errorSchema: error,
    key: "roles"
  }
});
