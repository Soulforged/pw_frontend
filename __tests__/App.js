//@flow
import React from "react";
import ReactDOM from "react-dom";
import App from "src/App";
import createStore from "src/reduxconf";
import createHistory from "history/createBrowserHistory";
import renderer from "react-test-renderer";
import actions from "src/actions";

const { LOGIN_SUCCESS } = actions;

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("shows login when not logged in", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("shows home (dashboard) when logged in", () => {
  const history = createHistory();
  const store = createStore({ history });
  store.dispatch({ type: LOGIN_SUCCESS, response: { details: {} } });
  const rendered = renderer.create(<App history={history} store={store} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
