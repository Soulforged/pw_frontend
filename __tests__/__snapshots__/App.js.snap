// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows home (dashboard) when logged in 1`] = `
<div>
  <div
    className="animate"
  >
    <div
      id="wrapper"
    >
      <div
        className="text-center animate"
        id="wrapper_left"
        style={
          Object {
            "overflow": "hidden",
            "width": "14%",
          }
        }
      >
        <div
          className="left-relative"
        >
          <div
            className="left-up"
          >
            <div
              className="text-right"
            >
              <i
                aria-hidden="true"
                className="menu-toggle fa fa-bars pointer"
                onClick={[Function]}
              />
            </div>
            <div
              className="logo-big"
            >
              <img
                alt="logo"
                className="home pointer"
                src="pegb-logo-final-white.svg"
              />
            </div>
          </div>
          <div
            className="left-center"
          >
            <div
              className="usrnm-lbl"
            >
              <div
                className="usrnm-big"
              >
                <span />
                <button
                  className="fa fa-chevron-down pointer icon-button"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>
          <div
            className="left-down"
          >
            <ul
              className="menu-nav"
            >
              <li
                className="nav-tabs"
              >
                <a
                  aria-current={false}
                  className={undefined}
                  href="/dashboard"
                  onClick={[Function]}
                  style={undefined}
                >
                  <img
                    alt="dashboard"
                    src="dashboard.svg"
                  />
                  <span
                    style={
                      Object {
                        "display": "inline",
                      }
                    }
                  >
                    Dashboard
                  </span>
                </a>
              </li>
              <li
                className="nav-tabs"
              >
                <a
                  aria-current={false}
                  className={undefined}
                  href="/users"
                  onClick={[Function]}
                  style={undefined}
                >
                  <img
                    alt="users"
                    src="admin-icon.svg"
                  />
                  <span
                    style={
                      Object {
                        "display": "inline",
                      }
                    }
                  >
                    Users
                  </span>
                </a>
              </li>
              <li
                className="nav-tabs"
              >
                <a
                  aria-current={false}
                  className={undefined}
                  href="/roles"
                  onClick={[Function]}
                  style={undefined}
                >
                  <img
                    alt="roles"
                    src="admin-icon.svg"
                  />
                  <span
                    style={
                      Object {
                        "display": "inline",
                      }
                    }
                  >
                    roles
                  </span>
                </a>
              </li>
              <li
                className="nav-tabs"
              >
                <a
                  aria-current={false}
                  className={undefined}
                  href="/business_units"
                  onClick={[Function]}
                  style={undefined}
                >
                  <img
                    alt="business_units"
                    src="admin-icon.svg"
                  />
                  <span
                    style={
                      Object {
                        "display": "inline",
                      }
                    }
                  >
                    Business Units
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <div
            className="absolute-down"
          >
            <div
              className="theme_toggle"
            >
              <label
                className="switch"
                htmlFor="theme_toggle"
              >
                <input
                  checked={false}
                  id="theme_toggle"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="slider round"
                />
              </label>
            </div>
            <p
              className="help-support"
              style={
                Object {
                  "display": "block",
                }
              }
            >
              <a
                href="/"
              >
                Help and support
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="animate"
        id="wrapper_right"
        style={
          Object {
            "overflow": "hidden",
            "width": "86%",
          }
        }
      >
        <div
          className="animate"
        >
          <div
            id="dashboard"
          >
            <div
              className="dash-search-pnl text-left"
              id="search-pnl"
            >
              <div
                id="filter-row"
              >
                <form
                  className="row"
                  onSubmit={[Function]}
                >
                  <div
                    className="col-md-4 col-sm-4"
                  >
                    <label
                      htmlFor="dateFrom"
                    >
                      Date from
                      <input
                        className="form-control"
                        name="dateFrom"
                        placeholder="Date From"
                        type="date"
                      />
                    </label>
                  </div>
                  <div
                    className="col-md-4 col-sm-4"
                  >
                    <label
                      htmlFor="dateTo"
                    >
                      Date to
                      <input
                        className="form-control"
                        name="dateTo"
                        placeholder="Date To"
                        type="date"
                      />
                    </label>
                  </div>
                  <div
                    className="col-md-3 col-sm-3"
                  />
                  <div
                    className="col-md-1 col-sm-1"
                  >
                    <button
                      className="btn filter-btn"
                      type="submit"
                    >
                      G0
                    </button>
                  </div>
                </form>
              </div>
            </div>
            <div
              className="tnx-flex"
            >
              <div
                className="tnx text-center col-md-3 col-sm-4 tnx-abs"
                id="tnx-avg"
              >
                <p
                  className="bold"
                >
                  SUMMARY
                </p>
                <div
                  className="tnx-sm cash-in"
                >
                  <div
                    id="loader"
                    style={Object {}}
                  >
                    <img
                      alt="loading something..."
                      className="loader"
                      src="loader.gif"
                    />
                  </div>
                  <span />
                  <p>
                    <span>
                      Cash-In Total Value
                    </span>
                  </p>
                </div>
                <div
                  className="tnx-sm cash-in"
                >
                  <div
                    id="loader"
                    style={Object {}}
                  >
                    <img
                      alt="loading something..."
                      className="loader"
                      src="loader.gif"
                    />
                  </div>
                  <span />
                  <p>
                    <span>
                      Cash-In Total Count
                    </span>
                  </p>
                </div>
                <div
                  className="tnx-sm cash-in"
                >
                  <div
                    id="loader"
                    style={Object {}}
                  >
                    <img
                      alt="loading something..."
                      className="loader"
                      src="loader.gif"
                    />
                  </div>
                  <span />
                  <p>
                    <span>
                      Cash-In Unique Customers
                    </span>
                  </p>
                </div>
                <div
                  className="tnx-sm cash-out"
                >
                  <div
                    id="loader"
                    style={Object {}}
                  >
                    <img
                      alt="loading something..."
                      className="loader"
                      src="loader.gif"
                    />
                  </div>
                  <span />
                  <p>
                    <span>
                      Cash-Out Total Value
                    </span>
                  </p>
                </div>
                <div
                  className="tnx-sm cash-out"
                >
                  <div
                    id="loader"
                    style={Object {}}
                  >
                    <img
                      alt="loading something..."
                      className="loader"
                      src="loader.gif"
                    />
                  </div>
                  <span />
                  <p>
                    <span>
                      Cash-Out Total Count
                    </span>
                  </p>
                </div>
                <div
                  className="tnx-sm cash-out"
                >
                  <div
                    id="loader"
                    style={Object {}}
                  >
                    <img
                      alt="loading something..."
                      className="loader"
                      src="loader.gif"
                    />
                  </div>
                  <span />
                  <p>
                    <span>
                      Cash-Out Unique customers
                    </span>
                  </p>
                </div>
                <div
                  className="clearfix"
                />
                <div
                  className="tnx-footer"
                />
              </div>
              <div
                className="text-center col-md-9 col-sm-8"
                id="tnx-graph"
              >
                <div
                  className="tnx-ema"
                >
                  <ul
                    className="nav nav-tabs"
                  >
                    <li
                      className="active"
                    >
                      <span>
                        Transaction Amount
                      </span>
                    </li>
                    <li>
                      <span>
                        Transaction Count
                      </span>
                    </li>
                  </ul>
                  <div
                    className="tab-content"
                  >
                    <div
                      className="tab-pane active"
                      id="trend-amount"
                    >
                      <div
                        className="tnx"
                      >
                        <div
                          id="loader"
                          style={
                            Object {
                              "height": "398px",
                            }
                          }
                        >
                          <img
                            alt="loading something..."
                            className="loader"
                            src="loader.gif"
                          />
                        </div>
                        <div
                          className="tnx-footer"
                          id="trendAmountChartLegend"
                        />
                      </div>
                    </div>
                    <div
                      className="tab-pane fade"
                      id="trend-count"
                    >
                      <div
                        className="tnx"
                      >
                        <div
                          className="tnx-footer"
                          id="trendCountChartLegend"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="tnx-flex tnx-max"
                >
                  <div
                    className="tnx-lg"
                  >
                    <ul
                      className="nav nav-tabs"
                    >
                      <li
                        className="active"
                      >
                        Cash-In
                      </li>
                      <li>
                        Cash-Out
                      </li>
                    </ul>
                    <div
                      className="tab-content"
                    >
                      <div
                        className="tab-pane active"
                        id="institutions-in"
                      >
                        <div
                          className="tnx"
                        >
                          <div
                            id="loader"
                            style={
                              Object {
                                "height": "289px",
                              }
                            }
                          >
                            <img
                              alt="loading something..."
                              className="loader"
                              src="loader.gif"
                            />
                          </div>
                          <div
                            className="tnx-footer"
                            id="cashinChartLegend"
                          />
                        </div>
                      </div>
                      <div
                        className="tab-pane fade"
                        id="institutions-out"
                      >
                        <div
                          className="tnx"
                        >
                          <canvas
                            id="cashoutChart"
                          />
                          <div
                            className="tnx-footer"
                            id="cashoutChartLegend"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="tnx-lg last"
                  >
                    <ul
                      className="nav nav-tabs"
                    >
                      <li
                        className="active"
                      >
                        Exceptions
                      </li>
                    </ul>
                    <div
                      className="tab-content"
                    >
                      <div
                        className="tab-pane active"
                        id="exceptions"
                      >
                        <div
                          className="tnx"
                        >
                          <p
                            className="bold"
                          >
                            EXCEPTIONS
                          </p>
                          <div
                            id="loader"
                            style={Object {}}
                          >
                            <img
                              alt="loading something..."
                              className="loader"
                              src="loader.gif"
                            />
                          </div>
                          <div
                            className="tnx-footer"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="clearfix"
                />
              </div>
            </div>
            <div
              className="clearfix"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`shows login when not logged in 1`] = `
<div>
  <div
    className="animate"
  >
    <div
      id="login-wrapper"
    >
      <div
        id="login-frame"
      >
        <div
          className="pnls"
          id="login"
        >
          <div
            className="login-header"
          >
            <p
              className="login-title color-6"
            >
              BACK OFFICE
            </p>
          </div>
          <form
            id="login_frm"
            onSubmit={[Function]}
          >
            <div
              className="login-body"
            >
              <div
                className="modal-p"
              >
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                  >
                    <span
                      className="login-user"
                    />
                  </span>
                  <input
                    className="form-control"
                    id="username"
                    onChange={[Function]}
                    placeholder="User@domain.com"
                    required={true}
                    type="text"
                  />
                </div>
              </div>
              <div
                className="modal-p"
              >
                <div
                  className="input-group"
                >
                  <span
                    className="input-group-addon"
                  >
                    <span
                      className="login-pswd"
                    />
                  </span>
                  <input
                    className="form-control"
                    id="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                  />
                </div>
              </div>
            </div>
            <div
              className="login-footer"
            >
              <div
                className="modal-p"
              >
                <input
                  className="btn btn-md"
                  type="submit"
                  value="LOGIN"
                />
              </div>
              <div
                className="modal-p"
              >
                <a
                  className="forgot-pswd color-6"
                  href="/forgotPassword"
                >
                  Forgot your password?
                </a>
              </div>
              <div
                className="row color-6"
              >
                <div
                  className="col-md-6 col-sm-6 text-left"
                >
                  Powered by Peg b
                </div>
                <div
                  className="col-md-6 col-sm-6 text-right"
                >
                  <i
                    className="logo-s"
                  />
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
